packages/omezarr/src/zarr/cached-loading/store.ts

```diff
@@ -0,0 +1,288 @@
+import { type Cacheable, logger, PriorityCache } from '@alleninstitute/vis-core';
+import * as zarr from 'zarrita';
+import { WorkerPool } from './worker-pool';
+import {
+    FETCH_SLICE_MESSAGE_TYPE,
+    type FetchSliceResponseMessage,
+    isFetchSliceResponseMessage,
+} from './fetch-slice.interface';
+
+const DEFAULT_NUM_WORKERS = 6;
+const DEFAULT_MAX_DATA_CACHE_BYTES = 256 * 2 ** 10; // 256 MB -- aribtrarily chosen at this point
+
+// @TODO implement a much more context-aware cache size limiting mechanism
+const getDataCacheSizeLimit = () => {
+    return DEFAULT_MAX_DATA_CACHE_BYTES;
+};
+
+export const asCacheKey = (key: zarr.AbsolutePath, range?: zarr.RangeQuery | undefined): string => {
+    const keyStr = JSON.stringify(key);
+    const rangeStr = range ? JSON.stringify(range) : 'no-range';
+    return `${keyStr} ${rangeStr}`;
+};
+
+type FetchStoreOptions = {
+    overrides?: RequestInit;
+    useSuffixRequest?: boolean;
+};
+
+class CacheableByteArray implements Cacheable {
+    #arr: Uint8Array;
+
+    constructor(arr: Uint8Array) {
+        this.#arr = arr;
+    }
+
+    destroy() {}
+
+    sizeInBytes(): number {
+        return this.#arr?.byteLength ?? 0;
```

Comment:
hmm - not clear to me how arr could be undefined here - ah perhaps this is leftovers from a refactor here?